crates/cargo-contract/src/cmd/call.rs

```diff
@@ -0,0 +1,249 @@
+use crate::ErrorVariant;
+
+use contract_build::util::DEFAULT_KEY_COL_WIDTH;
+use std::fmt::Debug;
+
+use super::{
+    display_contract_exec_result,
+    display_contract_exec_result_debug,
+    display_dry_run_result_warning,
+    print_dry_running_status,
+    print_gas_required_success,
+    prompt_confirm_tx,
+    CLIExtrinsicOpts,
+    MAX_KEY_COL_WIDTH,
+};
+use anyhow::{
+    anyhow,
+    Context,
+    Result,
+};
+use contract_build::name_value_println;
+use contract_extrinsics::{
+    BalanceVariant,
+    CallCommandBuilder,
+    CallExec,
+    DefaultConfig,
+    ExtrinsicOptsBuilder,
+    StorageDeposit,
+    TokenMetadata,
+};
+use contract_transcode::Value;
+use sp_weights::Weight;
+use subxt::Config;
+
+#[derive(Debug, clap::Args)]
+#[clap(name = "call", about = "Call a contract")]
+pub struct CallCommand {
+    /// The address of the the contract to call.
+    #[clap(name = "contract", long, env = "CONTRACT")]
+    contract: <DefaultConfig as Config>::AccountId,
+    /// The name of the contract message to call.
+    #[clap(long, short)]
+    message: String,
+    /// The arguments of the contract message to call.
+    #[clap(long, num_args = 0..)]
+    args: Vec<String>,
+    #[clap(flatten)]
+    extrinsic_cli_opts: CLIExtrinsicOpts,
+    /// Maximum amount of gas (execution time) to be used for this command.
+    /// If not specified will perform a dry-run to estimate the gas consumed for the
+    /// call.
+    #[clap(name = "gas", long)]
+    gas_limit: Option<u64>,
+    /// Maximum proof size for this call.
+    /// If not specified will perform a dry-run to estimate the proof size required for
+    /// the call.
+    #[clap(long)]
+    proof_size: Option<u64>,
+    /// The value to be transferred as part of the call.
+    #[clap(name = "value", long, default_value = "0")]
+    value: BalanceVariant,
+    /// Export the call output in JSON format.
+    #[clap(long, conflicts_with = "verbose")]
+    output_json: bool,
+}
+
+impl CallCommand {
+    /// Returns whether to export the call output in JSON format.
+    pub fn output_json(&self) -> bool {
+        self.output_json
+    }
+}
+pub async fn handle_call(call_command: &CallCommand) -> Result<(), ErrorVariant> {
```

Comment:
Not a huge thing but these `handle_` methods could be inherent methods on their respective commands. e.g. 

``rust
impl CallCommand {
  pub async fn handle(&self) { .. }
}
```